modules/nf-core/iphop/predict/main.nf

```diff
@@ -25,17 +25,28 @@ process IPHOP_PREDICT {
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
     export PERL5LIB=/usr/local/lib/perl5/site_perl/5.22.0
+
+    INPUT=$fasta
+    if [[ $fasta == *.gz ]]
+    then
+        gunzip -c $fasta > ${prefix}.fasta
+        INPUT=${prefix}.fasta
+    fi
+
     iphop \\
         predict \\
-        --fa_file $fasta \\
+        --fa_file \$INPUT \\
         --out_dir iphop_results \\
         --db_dir $iphop_db \\
         --num_threads $task.cpus \\
         $args
 
-    mv iphop_results/Host_prediction_to_genus_m*.csv .
-    mv iphop_results/Host_prediction_to_genome_m*.csv .
-    mv iphop_results/Detailed_output_by_tool.csv .
+    for FILE in iphop_results/Host_prediction_to_*.csv; do
+        mv \${FILE} ./${prefix}.\${FILE##*/}
+    done
+
+    mv iphop_results/Detailed_output_by_tool.csv ./${prefix}.Detailed_output_by_tool.csv
+    rm ${prefix}.fasta
```

Comment:
This is only present if it was unzipped, so you'll get a failure if not.

Reply:
Do you think rm -f would be ok? So, it will delete the file if it exists and otherwise just pass.
I would prefer that instead of adding more lines to see whether the file was zipped/plain.

Reply:
Hmm. That seems to return an exitcode of 0, so should be fine?